src/main/java/com/github/swrirobotics/bags/BagController.java

```diff
@@ -60,35 +60,30 @@
 @RequestMapping("bags")
 @ControllerAdvice
 public class BagController {
-    @Autowired
-    private BagService myBagService;
+    private final BagService myBagService;
 
     private final Logger myLogger = LoggerFactory.getLogger(BagController.class);
 
+    public BagController(BagService myBagService) {
+        this.myBagService = myBagService;
+    }
+
     @RequestMapping(value="/download", produces="application/x-bag")
-    public FileSystemResource downloadBag(
+    public InputStreamResource downloadBag(
             @RequestParam String bagId,
             HttpServletResponse response) throws IOException {
-        Long id = Long.valueOf(bagId);
+        long id = Long.parseLong(bagId);
```

Comment:
My java is really rusty. Is the change from ```Long``` to ```long``` significant here?

Reply:
The difference between them is that a `Long` can be `null` but a `long` can't be.  Because the `bagId` param on the function isn't optional, it will never be null, so this is an insignificant performance improvement.  (I changed it mostly just because my IDE warned me about it)